lib/elasticsearch_client.rb

```diff
@@ -0,0 +1,55 @@
+require 'elasticsearch'
+
+class ElasticsearchClient
```

Comment:
my preference would be that we name this to be more specific to us.  it's not really a ES client (that's in the Elasticsearch ruby integration gem).  it's OSElasticSearchClient (this helps to state that it's our internal one and that we use the Elasticsearch::Client (it's clear that this is the gem's version).

Reply:
Also, all this could be a bit simpler if you want.... e.g., this is what i did in my other project:

`initializers/elasticsearch.rb`
```
require 'faraday_middleware/aws_signers_v4'

elasticsearch_params = {
  log: true,
  url: ENV.fetch('AWS_ELASTICSEARCH_URL')
}

OpenSearch::ElasticSearchClient = Elasticsearch::Client.new(elasticsearch_params) do |f|
  f.request(
    :aws_signers_v4,
    credentials: Aws::Credentials.new(ENV.fetch('AWS_ACCESS_KEY'), ENV.fetch('AWS_SECRET_ACCESS_KEY')),
    service_name: 'es',
    region: ENV.fetch('AWS_REGION')
  )
end
```

Reply:
Making it `OpenSearch::ElasticsearchClient` works for me.

I don’t know if we can simplify the code quite this much because we have three kinds of configurations to handle: 1) local development where ES is in a docker container and requests aren’t signed, 2) test development where we sign requests using key and secret like you have above, and 3) production use where we use the built-in credentials on EC2 instances.  The wrapper class in this PR lets us do all three of these and lets us control the endpoint via a secret.  Does you agree?

Incidentally, I saw the client definition in your PR was in `config/initializers` but I have not typically put class definitions in that folder (instead used it for configuration calls), which is why I put this one in `lib`.  

(separately I thought I had to use `patron` is that not your experience?)

Reply:
class definitions in initializers are fine, or the global i used ^.  Not sure it matters that much. I just like that stuff that is "initialized" is there (perhaps it's using a class defined in lib).   I tend to think of things in lib as stuff that you would eventually convert to a gem (possibly).   

the wrapper class you wrote is good.  i just tossed this out there in case to consider a simpler idea, if possible...but it sounds like not the case. 

I've used the above ^ code in my other project working with ES both locally (unsigned, locally installed) and in production (signed, AWS ES).  it works for both (the local version just ignores the credentials, and uses a localhost:9200 url)